### ISSUE TITLE
[Logic Bug] A Sepcial SQL Brings  Errors ( SELECT * FROM <table> WHERE ( (<column> is null )::INT IS NOT DISTINCT FROM <column>) OR <column> )

### ISSUE THREAD

**Issue by qwebug** [heterodb/pg-strom#650]
#### Describe:

&#x20;Using SELECT \* FROM \<table> WHERE ( (\<column> is null )::INT IS NOT DISTINCT FROM \<column>) OR \<column>  brings different results, when turn on and off pg\_strom.enabled.

#### ***Sql:***

```sql
CREATE TABLE t0(c0 REAL);
CREATE TABLE t1(c0 boolean);
INSERT INTO t1(c0) VALUES(null);
INSERT INTO t0(c0) VALUES(0.1);
CREATE SCHEMA extensions;
CREATE EXTENSION pg_strom WITH SCHEMA extensions;
set pg_strom.enabled=on;
SELECT * FROM t0,t1 WHERE ((t0.c0)::INT IS NOT DISTINCT FROM (t0.c0)) OR (t1.c0);
```

#### ***Result:***

```bash
 c0  | c0 
-----+----
 0.1 | 
(1 row)
```

#### ***Sql:***

```sql
set pg_strom.enabled=off;
SELECT * FROM t0,t1 WHERE ((t0.c0)::INT IS NOT DISTINCT FROM (t0.c0)) OR (t1.c0);
```

#### ***Result:***

```bash
 c0 | c0 
----+----
(0 rows)
```

### Environment:

Pg-strom Version: commit b959fef3547bb9050d8159bb2d8cd4aa4dc918d5

PostgreSQL Version: 15.3 on x86\_64-pc-linux-gnu, compiled by gcc (GCC) 8.5.0 20210514 (Red Hat 8.5.0-18), 64-bit

CUDA Version: 12.0.1

NVIDIA Driver Version: 535.86.05

**Comment by kaigai** [comment]
This query uses NOT operator, and it had a problem that was solved at `7a2ab9c1c1658367f6685caa21f68d2a5928dd65` (bugfix: BoolExprNot referenced uninitialized variable).
It seems to me the identical problem with #647, and not reproduced at the latest commit.

**Comment by qwebug** [comment]
This problem is fixed, after my verification.
Thanks for your work.
